src/xlat-i/xi-Parameter.C

```diff
@@ -65,7 +65,7 @@ Parameter::Parameter(int Nline, Type* Ntype, const char* Nname, const char* Narr
   if (name == NULL && !isVoid()) { /*Fabricate a unique name for this marshalled param.*/
     static int unnamedCount = 0;
     name = new char[50];
-    sprintf((char*)name, "impl_noname_%x", unnamedCount++);
+    snprintf((char*)name, 50, "impl_noname_%x", unnamedCount++);
```

Comment:
`static constexpr int len = 50;`

Reply:
We can't use C++11 or newer in charmxi until issue #3622 is fixed: https://github.com/UIUC-PPL/charm/issues/3622